src/rules/comment.ts

```diff
@@ -0,0 +1,124 @@
+import { createRule } from '../util';
+
+export const ruleName = 'comment-syntax';
+
+type Options = [];
+
+const SPACE_CHARCODE = 32;
+const NEWLINE_CHARCODE = 10;
+const STAR_CHARCODE = 42;
+
+type MessageIds =
+  | 'lineCommentCapital'
+  | 'lineCommentEnding'
+  | 'blockCommentCapital'
+  | 'blockCommentEnding'
+  | 'shouldStartWithSpace'
+  | 'shouldStartWithBlock';
+
+export default createRule<Options, MessageIds>({
+  name: ruleName,
+  meta: {
+    type: 'layout',
+    docs: {
+      category: 'Best Practices',
+      description: `Enforce block comments to have capital first letter and end with dots and 
+      line comments no capital first letter and no dot`,
+      recommended: 'error',
+    },
+    fixable: 'code',
+    schema: [],
+    messages: {
+      shouldStartWithSpace: 'A line comment should start with a space',
+      shouldStartWithBlock: 'A block comment should start with /**\n *',
+      lineCommentCapital: 'A line comment cannot start with a capital letter unless the entire word is',
+      lineCommentEnding: 'A line comment cannot end with a dot',
+      blockCommentCapital: 'A block comment has to start with a capital letter',
+      blockCommentEnding: 'A block comment has to end with a dot',
+    },
+  },
+  defaultOptions: [],
+  create: (context) => {
+    const source = context.getSourceCode();
+
+    const comments = source.getAllComments();
+
+    const isCapital = (char: string) => {
+      return char !== char.toLowerCase();
```

Comment:
Random nitpick, no need to change anything:
I find strange that you use `toLowerCase` to check for upper-case-ness instead of `char === char.toUpperCase()`.

Reply:
😂 
Valid point.